dcpy/lifecycle/package/validate.py

```diff
@@ -211,95 +203,110 @@ def validate_df(
                     ValidationError(
                         error_type=ErrorType.INVALID_DATA,
                         message=f"Found counts of non-standardized values for column {col.name}: {invalid_counts_int}",
-                        dataset_file=dataset,
                     )
                 )
 
         # Check Nulls
-        if col.non_nullable:
+        if col.checks and col.checks.non_nullable:
             if not df_only_col_nulls.empty:
                 errors.append(
                     ValidationError(
                         error_type=ErrorType.NULLS_FOUND,
                         message=f"Column {col.name} has {len(df_only_col_nulls)} empty values",
-                        dataset_file=dataset,
                     )
                 )
 
-    return DatasetFileValidation(
-        errors=errors, stats=ValidationStats(row_count=len(df))
-    )
+    return errors
 
 
 def validate_csv(
-    csv_path: Path, dataset: models.DatasetFile, metadata: models.Metadata
-) -> DatasetFileValidation:
+    csv_path: Path, columns: list[dataset_md.DatasetColumn]
+) -> list[ValidationError]:
     df = pd.read_csv(csv_path, dtype=str)
-    return validate_df(df, dataset, metadata)
+    return validate_df(df, columns)
 
 
 def validate_shapefile(
-    shp_path: Path, dataset: models.DatasetFile, metadata: models.Metadata
```

Comment:
If these are becoming the same, we could use one of our utils to read in df/gdf from filepath - then this just becomes one "validate_file" function.

I don't actually think we're quite there though - `read_data_to_df` in `utils/data.py` is a little more tailored to ingest. But it'd be nice to align that sort of logic with this metadata logic

Reply:
That's very true. Added a note about [follow-on work](https://github.com/orgs/NYCPlanning/projects/27/views/4?filterQuery=is%3Aissue++-project%3A*%5Bbacklog%5D*&pane=issue&itemId=68576094) for when we swap in Pandera